File: DSA/Java/test/dsa/lib/algo/sort/SelectionSortTest.java
package dsa.lib.algo.sort;

import org.junit.Test;

import static org.junit.Assert.*;

public class SelectionSortTest {
    @Test
    public void oneNum() {
        int[] ary = {1};
        final int[] expected = {1};
        SelectionSort.sort(ary);
        assertArrayEquals(expected, ary);
    }

    @Test
    public void case1() {
        int[] ary = {4, 5, 6, 3, 2, 1};
        final int[] expected = {1, 2, 3, 4, 5, 6};
        SelectionSort.sort(ary);
        assertArrayEquals(expected, ary);
    }

}

File: DSA/Cpp/test/dsa/lib/algo/sort/selection_sort_test.cpp
#include <dsa/lib/algo/sort/selection_sort.hpp>
#include "sort_test.h"

namespace dsa::lib::algo::sort {
namespace {

class SelectionSortTest : public SortTest {};
TEST_F(SelectionSortTest, oneNum) {
    int ary[] = {1};
    int expected[] = {1};
    SelectionSort::sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}
TEST_F(SelectionSortTest, case1) {
    int ary[] = {4, 5, 6, 3, 2, 1};
    int expected[] = {1, 2, 3, 4, 5, 6};
    SelectionSort::sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}

}
};